### DISCUSSION TITLE
how to analyze coefficient of variation in expression and methylation

### DISCUSSION THREAD

**Discussion by shellywanamaker** [sr320/ceabigr#61] [Q&A]
@yaaminiv and I attempted a first pass at [CoV analysis](https://github.com/sr320/ceabigr/blob/main/code/CoV_exp_x_meth.Rmd). Our initial attempt was to see if treatment impacted expression variation within each sex (log(expression CoV) ~ treatment) to better understand the expression PCAs that @laurahspencer shared at a previous meeting. So far we found a significant difference P < 2*10^-16 of treatment on CoV for males and females, but the distributions of CoV for each treatment do not appear that different.

![image](https://user-images.githubusercontent.com/4720221/170108731-6cb909f4-2261-4022-9a52-086adc070771.png)

here's the boxplots for females:
![image](https://user-images.githubusercontent.com/4720221/170110200-95af9443-3665-4ad2-b4a2-8f4171b97dee.png)

Does anyone have ideas for an alternate analysis with checking if expression CoV is impacted by treatment alone within each sex?

Some more integrative approaches we are considering the analyses done in this wasp paper https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1008397  and the paper laura posted in the manuscript https://www.google.com/url?q=https://journals.physiology.org/doi/full/10.1152/physiolgenomics.00128.2018&sa=D&source=docs&ust=1653420243427699&usg=AOvVaw15biWhkSEaOrslv7zgJwaN

**Comment by sr320** [comment]
to confirm the is based on gene level expression?

I think it would be good to look at variation exons within genes, or possibly transcript variation.

**Reply by shellywanamaker** [reply]
yes, gene level. We can try exon and transcript level too

**Comment by shellywanamaker** [comment]
@laurahspencer can you share you PCA loadings per gene data so we can check the CoV distribution for the genes with the highest loadings magnitude ?

**Reply by laurahspencer** [reply]
Yep- I'll get that to you by the end of this week!

**Reply by shellywanamaker** [reply]
@laurahspencer what was the input for the PCAs? FPKM? Do you think you would get different PCA results if you use FPKM, TPM, or TMM?

**Reply by laurahspencer** [reply]
Per-gene PC loadings for the egg-only PCA: https://github.com/sr320/ceabigr/blob/main/output/LS_PCA-eggs_gene-loadings_fpkm.csv. These were extracted from the below PCA, which I generated using `prcomp()`. 

Input was the gene-wise FPKM matrix ([gene_fpkm.csv](https://github.com/sr320/ceabigr/blob/main/data/gene_fpkm.csv)), which I filtered to remove low-frequency genes (those with <10 summed across all samples). Only the first 2 PCs are significant as per the screeplot.  I used ANOVA to compare the euclidean distances from each sample to its per-treatment centroid, and they tend to differ by treatment (F=3.5, p=0.081). I also tested for multivariate homogeneity of variance using `adonis()` and it does not indicate significant differences among treatments.  

Code for this work is lines 118-204 in this notebook: [gene-expression-analysis.Rmd](https://github.com/sr320/ceabigr/blob/main/code/gene-expression-analysis.Rmd)

![image](https://user-images.githubusercontent.com/17264765/172309445-fac4f541-fde8-43fa-bd21-4af480bcd69f.png)

**Comment by shellywanamaker** [comment]
Another thought brought up by @yaaminiv from [this paper](https://www.google.com/url?q=https://journals.physiology.org/doi/full/10.1152/physiolgenomics.00128.2018&sa=D&source=docs&ust=1653420243427699&usg=AOvVaw15biWhkSEaOrslv7zgJwaN):

Because reads are normalized across samples and genes (e.g. to FPKM) should we just be looking at variance or is coefficient of variation still appropriate? Maybe this doesn't really matter if they are proportional. 

Should we be using TPM instead of FPKM?

**Reply by yaaminiv** [reply]
@kubu4 Do you have any thoughts on TPM vs FPKM? Is there a way to get TPM output from `ballgown`/does TPM for this dataset exist anywhere?

**Reply by shellywanamaker** [reply]
These are plots of the coefficient of variation from FPKM values for 20 genes with the highest PC1 and PC2 loadings and lowest PC1 and PC2 loadings. Each dot represents the CoV for a gene. These plots of these subsetted genes appear to have a greater difference in variation between control and treated group that is not apparent when looking at all genes (black and white boxplots above).


![image](https://user-images.githubusercontent.com/4720221/172442379-04dfd2e9-cb9b-43d0-b61d-d382f655290f.png)

**Reply by laurahspencer** [reply]
Here is one set of genes that are top contributors to PC1 

![image](https://user-images.githubusercontent.com/17264765/172448814-53f3755f-865d-4815-b975-dc800795d756.png)

**Reply by kubu4** [reply]
> Do you have any thoughts on TPM vs FPKM? Is there a way to get TPM output from ballgown/does TPM for this dataset exist anywhere?

@yaaminiv - Does [this paper](https://translational-medicine.biomedcentral.com/articles/10.1186/s12967-021-02936-w) "over complicate" the issue? It's a comparison of TPM vs FPKM vs count normalization (e.g. DEseq2) and using these different values to compare across replicate samples. TL;DR - paper suggests that TPM is the worst performer for cross-sample comparisons; FPKM is okay, but recommendation is to use count normalization.


Regardless, I'll look into `ballgown` a bit more and see what's doing!
